pkg/apis/flowcontrol/types.go

```diff
@@ -0,0 +1,278 @@
+	// exempt priority level would be the highest priority level. The field is default to false and only those system
+	// preset priority level can be exempt.
+	// +optional
+	Exempt bool
+	// AssuredConcurrencyShares is a positive number for a non-exempt priority level, representing the weight by which
+	// the priority level shares the concurrency from the global limit. The concurrency limit of an apiserver is divided
+	// among the non-exempt priority levels in proportion to their assured concurrency shares. Basically this produces
+	// the assured concurrency value (ACV) for each priority level:
+	//
+	//             ACV(l) = ceil( SCL * ACS(l) / ( sum[priority levels k] ACS(k) ) )
+	//
+	AssuredConcurrencyShares int32
+	// Queues is a number of queues that belong to a non-exempt PriorityLevelConfiguration object. The queues exist independently
+	// at each apiserver.
+	Queues int32
+	// HandSize is a small positive number for applying shuffle sharding. When a request arrives at an apiserver the
+	// request flow identifier’s string pair is hashed and the hash value is used to shuffle the queue indices and deal
+	// a hand of the size specified here. If empty, the hand size will the be set to 1 at which  the priority drains the
+	// queues on a FIFO basis.
+	// +optional
+	HandSize int32
+	// QueueLengthLimit is a length limit applied to each queue belongs to the priority.
+	// +optional
+	QueueLengthLimit int32
+}
+
+// PriorityLevelConfigurationConditionType is a valid value for PriorityLevelConfigurationStatusCondition.Type
+type PriorityLevelConfigurationConditionType string
+
+// PriorityLevelConfigurationConditionStatus is a valid value for PriorityLevelConfigurationStatusCondition.Status
+type PriorityLevelConfigurationConditionStatus string
+
+// PriorityLevelConfigurationStatus represents the current state of a request-priority.
+type PriorityLevelConfigurationStatus struct {
+	// Current state of request-priority.
+	Conditions []PriorityLevelConfigurationStatusCondition
+}
+
+// PriorityLevelConfigurationStatusCondition defines the condition of priority level.
+type PriorityLevelConfigurationStatusCondition struct {
+	// Type is the type of the condition.
+	Type PriorityLevelConfigurationConditionType
+	// Status is the status of the condition.
+	// Can be True, False, Unknown.
+	Status PriorityLevelConfigurationConditionStatus
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time
+	// Unique, one-word, CamelCase reason for the condition's last transition.
+	Reason string
+	// Human-readable message indicating details about last transition.
+	Message string
+}
+
+// Subject contains a reference to the object or user identities a role binding applies to.  This can either hold a direct API object reference,
+// or a value for non-objects such as user and group names.
+type Subject struct {
```

Comment:
We no longer use this type.

Reply:
removed

Reply:
I wonder if you have pushed all your updates.  I still see the `Subject` type defined here.